initial setup

0xtruly opened the pull request.

0xtruly committed:
initial setup

0xtruly committed:
Implement pagination
Update properties json and README

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
 
 const startApp = async () => {
-    
+    paginate();
```
Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  paginate();
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+        const data: ResultData[] = resData?.results[0][`${currentPage}`];
+        if (paging.previous) {
+            previousBtn.disabled = false;
+            previousBtn.style.color = "#04AA6D";
```
Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      previousBtn.style.color = "#04AA6D";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+type ResultData = {
+    id: string,
+    gender: string,
```
Codacy has a fix for the issue: [Replace `··gender:·string,` with `gender:·string;`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  gender: string;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+        await loadData(currentPage);
+    });
+    nextBtn.addEventListener('click', async () => {
+        currentPage++;
```
Codacy has a fix for the issue: [Replace `········` with `····`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    currentPage++;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+type ResultData = {
+    id: string,
```
Codacy has a fix for the issue: [Replace `··id:·string,` with `id:·string;`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  id: string;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+    const response = await fetch(`https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`);
+
+    if (response.status === 200) {
+        const resData = await response.json();
```
Codacy has a fix for the issue: [Delete `····`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    const resData = await response.json();
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+}
+
+type PagingData = {
+    next?: string,
```
Codacy has a fix for the issue: [Replace `··next?:·string,` with `next?:·string;`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  next?: string;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+    pageView.innerHTML = `Page: ${currentPage}`;
+    previousBtn.addEventListener('click', async () => {
+        if (currentPage > 1) {
+            currentPage--;
```
Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      currentPage--;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+
+type PagingData = {
+    next?: string,
+    previous?: string,
```
Codacy has a fix for the issue: [Replace `··previous?:·string,` with `previous?:·string;`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previous?: string;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+
+    if (response.status === 200) {
+        const resData = await response.json();
+        const paging: PagingData = resData?.results[0].paging;
```
Codacy has a fix for the issue: [Replace `········` with `····`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    const paging: PagingData = resData?.results[0].paging;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+            previousBtn.style.color = "#04AA6D";
+        }
+
+        for (let i = 0; i < data?.length; i++) {
```
Codacy has a fix for the issue: [Replace `········` with `····`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    for (let i = 0; i < data?.length; i++) {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+}
+
+let previousBtn: HTMLElement | any = document.querySelector('button[data-prevbtn]');
+let nextBtn: HTMLElement | any = document.querySelector('button[data-nextbtn]');
```
Codacy has a fix for the issue: [Replace `'button[data-nextbtn]'` with `"button[data-nextbtn]"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let nextBtn: HTMLElement | any = document.querySelector("button[data-nextbtn]");
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+        const paging: PagingData = resData?.results[0].paging;
+        const data: ResultData[] = resData?.results[0][`${currentPage}`];
+        if (paging.previous) {
+            previousBtn.disabled = false;
```
Codacy has a fix for the issue: [Delete `······`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      previousBtn.disabled = false;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+                <td id=${row.id}>${row.row}</td>
+                <td>${row.gender}</td>
+                <td>${row.age}</td>`;
+            tableBody.appendChild(tr);
```
Codacy has a fix for the issue: [Delete `······`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      tableBody.appendChild(tr);
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+
+
+const loadData = async (page: number = 1) => {
+    const response = await fetch(`https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`);
```
Codacy has a fix for the issue: [Replace `··const·response·=·await·fetch(`https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`` with `const·response·=·await·fetch(⏎····`https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`⏎··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  const response = await fetch(
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+
+const paginate = () => {
+    previousBtn.disabled = true;
+    previousBtn.style.color = "black";
```
Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.style.color = "black";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+type ResultData = {
+    id: string,
+    gender: string,
+    age: number,
```
Codacy has a fix for the issue: [Replace `····age:·number,` with `··age:·number;`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  age: number;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+    previousBtn.disabled = true;
+    previousBtn.style.color = "black";
+    previousBtn.style.backgroundColor = "gray";
+    pageView.innerHTML = `Page: ${currentPage}`;
```
Codacy has a fix for the issue: [Replace `····` with `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  pageView.innerHTML = `Page: ${currentPage}`;
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+    previousBtn.style.backgroundColor = "gray";
+    pageView.innerHTML = `Page: ${currentPage}`;
+    previousBtn.addEventListener('click', async () => {
+        if (currentPage > 1) {
```
Codacy has a fix for the issue: [Delete `····`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    if (currentPage > 1) {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
 const startApp = async () => {
-    
+    paginate();
+    await loadData();
```
Codacy has a fix for the issue: [Replace `····` with `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  await loadData();
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+        const resData = await response.json();
+        const paging: PagingData = resData?.results[0].paging;
+        const data: ResultData[] = resData?.results[0][`${currentPage}`];
+        if (paging.previous) {
```
Codacy has a fix for the issue: [Replace `········` with `····`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    if (paging.previous) {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+let previousBtn: HTMLElement | any = document.querySelector('button[data-prevbtn]');
+let nextBtn: HTMLElement | any = document.querySelector('button[data-nextbtn]');
+let pageView: HTMLElement | any = document.querySelector('label[data-pageview]');
+let tableBody: HTMLElement | any = document.querySelector('tbody[data-sink]');
```
Codacy has a fix for the issue: [Replace `'tbody[data-sink]'` with `"tbody[data-sink]"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let tableBody: HTMLElement | any = document.querySelector("tbody[data-sink]");
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+            currentPage--;
+        }
+        pageView.innerHTML = `Page: ${currentPage}`;
+        tableBody.innerHTML = '';
```
Codacy has a fix for the issue: [Replace `········tableBody.innerHTML·=·''` with `····tableBody.innerHTML·=·""`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    tableBody.innerHTML = "";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+    gender: string,
+    age: number,
+    row: number,
+}
```
Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
};
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,72 @@
+let currentPage: number = 1;
+
+const paginate = () => {
+    previousBtn.disabled = true;
```
Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.disabled = true;
```

0xtruly committed:
linting

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+  previous?: string;
+};
+
+let previousBtn: HTMLElement | any = document.querySelector("button[data-prevbtn]");
```
Codacy has a fix for the issue: [Replace `"button[data-prevbtn]"` with `⏎··"button[data-prevbtn]"⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let previousBtn: HTMLElement | any = document.querySelector(
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+
+const paginate = () => {
+  previousBtn.disabled = true;
+  previousBtn.style.color = 'black';
```
Codacy has a fix for the issue: [Replace `'black'` with `"black"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.style.color = "black";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+  nextBtn.addEventListener('click', async () => {
+    currentPage++;
+    pageView.innerHTML = `Page: ${currentPage}`;
+    tableBody.innerHTML = '';
```
Codacy has a fix for the issue: [Replace `········tableBody.innerHTML·=·''` with `····tableBody.innerHTML·=·""`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    tableBody.innerHTML = "";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+  previousBtn.style.color = 'black';
+  previousBtn.style.backgroundColor = 'gray';
+  pageView.innerHTML = `Page: ${currentPage}`;
+  previousBtn.addEventListener('click', async () => {
```
Codacy has a fix for the issue: [Replace `····previousBtn.addEventListener('click'` with `··previousBtn.addEventListener("click"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.addEventListener("click", async () => {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+    tableBody.innerHTML = "";
+    await loadData(currentPage);
+  });
+  nextBtn.addEventListener('click', async () => {
```
Codacy has a fix for the issue: [Replace `····nextBtn.addEventListener('click'` with `··nextBtn.addEventListener("click"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  nextBtn.addEventListener("click", async () => {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+const paginate = () => {
+  previousBtn.disabled = true;
+  previousBtn.style.color = 'black';
+  previousBtn.style.backgroundColor = 'gray';
```
Codacy has a fix for the issue: [Replace `'gray'` with `"gray"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.style.backgroundColor = "gray";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+    const data: ResultData[] = resData?.results[0][`${currentPage}`];
+    if (paging.previous) {
+      previousBtn.disabled = false;
+      previousBtn.style.color = '#04AA6D';
```
Codacy has a fix for the issue: [Replace `'#04AA6D'` with `"#04AA6D"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      previousBtn.style.color = "#04AA6D";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+
+    for (let i = 0; i < data?.length; i++) {
+      const row = data[i];
+      let tr = document.createElement('tr');
```
Codacy has a fix for the issue: [Replace `············let·tr·=·document.createElement('tr'` with `······let·tr·=·document.createElement("tr"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      let tr = document.createElement("tr");
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+
+const loadData = async (page: number = 1) => {
+  const response = await fetch(
+    `https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`,
```
Codacy has a fix for the issue: [Delete `,`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    `https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,71 @@
+
+let previousBtn: HTMLElement | any = document.querySelector("button[data-prevbtn]");
+let nextBtn: HTMLElement | any = document.querySelector("button[data-nextbtn]");
+let pageView: HTMLElement | any = document.querySelector("label[data-pageview]");
```
Codacy has a fix for the issue: [Replace `"label[data-pageview]"` with `⏎··"label[data-pageview]"⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let pageView: HTMLElement | any = document.querySelector(
```

0xtruly committed:
failing test

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+    const data: ResultData[] = resData?.results[0][`${currentPage}`];
+    if (paging.previous) {
+      previousBtn.disabled = false;
+      previousBtn.style.color = '#04AA6D';
```
Codacy has a fix for the issue: [Replace `'#04AA6D'` with `"#04AA6D"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      previousBtn.style.color = "#04AA6D";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+
+const paginate = () => {
+  previousBtn.disabled = true;
+  previousBtn.style.color = 'black';
```
Codacy has a fix for the issue: [Replace `'black'` with `"black"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.style.color = "black";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+    for (let i = 0; i < data?.length; i++) {
+      const row = data[i];
+      let tr = document.createElement('tr');
+      tr?.setAttribute('data-entryId', row.id);
```
Codacy has a fix for the issue: [Replace `'data-entryId'` with `"data-entryId"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      tr?.setAttribute("data-entryId", row.id);
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+const paginate = () => {
+  previousBtn.disabled = true;
+  previousBtn.style.color = 'black';
+  previousBtn.style.backgroundColor = 'gray';
```
Codacy has a fix for the issue: [Replace `'gray'` with `"gray"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.style.backgroundColor = "gray";
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+  previous?: string;
+};
+
+let previousBtn: HTMLElement | any = document.querySelector("button[data-prevbtn]");
```
Codacy has a fix for the issue: [Replace `"button[data-prevbtn]"` with `⏎··"button[data-prevbtn]"⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let previousBtn: HTMLElement | any = document.querySelector(
```

dekunleroy replied:
Codacy has a fix for the issue: [Replace `"button[data-prevbtn]"` with `⏎··"button[data-prevbtn]"⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let previousBtn: HTMLElement | any = document.querySelector(
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+
+const loadData = async (page: number = 1) => {
+  const response = await fetch(
+    `https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`,
```
Codacy has a fix for the issue: [Delete `,`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    `https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=${page}`
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+    tableBody.innerHTML = "";
+    await loadData(currentPage);
+  });
+  nextBtn.addEventListener('click', async () => {
```
Codacy has a fix for the issue: [Replace `····nextBtn.addEventListener('click'` with `··nextBtn.addEventListener("click"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  nextBtn.addEventListener("click", async () => {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+
+let previousBtn: HTMLElement | any = document.querySelector("button[data-prevbtn]");
+let nextBtn: HTMLElement | any = document.querySelector("button[data-nextbtn]");
+let pageView: HTMLElement | any = document.querySelector("label[data-pageview]");
```
Codacy has a fix for the issue: [Replace `"label[data-pageview]"` with `⏎··"label[data-pageview]"⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let pageView: HTMLElement | any = document.querySelector(
```

dekunleroy replied:
Codacy has a fix for the issue: [Replace `"label[data-pageview]"` with `⏎··"label[data-pageview]"⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
let pageView: HTMLElement | any = document.querySelector(
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+  previousBtn.style.color = 'black';
+  previousBtn.style.backgroundColor = 'gray';
+  pageView.innerHTML = `Page: ${currentPage}`;
+  previousBtn.addEventListener('click', async () => {
```
Codacy has a fix for the issue: [Replace `····previousBtn.addEventListener('click'` with `··previousBtn.addEventListener("click"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.addEventListener("click", async () => {
```

0xtruly committed:
linting

0xtruly committed:
linting

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+
+    for (let i = 0; i < data?.length; i++) {
+      const row = data[i];
+      let tr = document.createElement('tr');
```
Codacy has a fix for the issue: [Replace `············let·tr·=·document.createElement('tr'` with `······let·tr·=·document.createElement("tr"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      let tr = document.createElement("tr");
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+  previousBtn.style.color = "black";
+  previousBtn.style.backgroundColor = "gray";
+  pageView.innerHTML = `Page: ${currentPage}`;
+  previousBtn.addEventListener('click', async () => {
```
Codacy has a fix for the issue: [Replace `····previousBtn.addEventListener('click'` with `··previousBtn.addEventListener("click"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  previousBtn.addEventListener("click", async () => {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+    tableBody.innerHTML = "";
+    await loadData(currentPage);
+  });
+  nextBtn.addEventListener('click', async () => {
```
Codacy has a fix for the issue: [Replace `····nextBtn.addEventListener('click'` with `··nextBtn.addEventListener("click"`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  nextBtn.addEventListener("click", async () => {
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+  nextBtn.addEventListener('click', async () => {
+    currentPage++;
+    pageView.innerHTML = `Page: ${currentPage}`;
+    tableBody.innerHTML = '';
```
Codacy has a fix for the issue: [Replace `········tableBody.innerHTML·=·''` with `····tableBody.innerHTML·=·""`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    tableBody.innerHTML = "";
```

0xtruly committed:
linting

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,80 @@
+      const index = Number(n) + 1;
+      const item = data[n];
+      const tR = document.querySelector(`tbody > tr:nth-child(${index})`);
+      const firstChild = document.querySelector(`tbody > tr:nth-child(${index}) > td:nth-child(1)`);
```
Codacy has a fix for the issue: [Replace ``tbody·>·tr:nth-child(${index})·>·td:nth-child(1)`` with `⏎········`tbody·>·tr:nth-child(${index})·>·td:nth-child(1)`⏎······`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      const firstChild = document.querySelector(
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,80 @@
+      const tR = document.querySelector(`tbody > tr:nth-child(${index})`);
+      const firstChild = document.querySelector(`tbody > tr:nth-child(${index}) > td:nth-child(1)`);
+      const secondChild = document.querySelector(`tbody > tr:nth-child(${index}) > td:nth-child(2)`);
+      const thirdChild = document.querySelector(`tbody > tr:nth-child(${index}) > td:nth-child(3)`);
```
Codacy has a fix for the issue: [Replace ``tbody·>·tr:nth-child(${index})·>·td:nth-child(3)`` with `⏎········`tbody·>·tr:nth-child(${index})·>·td:nth-child(3)`⏎······`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      const thirdChild = document.querySelector(
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,80 @@
+      const item = data[n];
+      const tR = document.querySelector(`tbody > tr:nth-child(${index})`);
+      const firstChild = document.querySelector(`tbody > tr:nth-child(${index}) > td:nth-child(1)`);
+      const secondChild = document.querySelector(`tbody > tr:nth-child(${index}) > td:nth-child(2)`);
```
Codacy has a fix for the issue: [Replace ``tbody·>·tr:nth-child(${index})·>·td:nth-child(2)`` with `⏎········`tbody·>·tr:nth-child(${index})·>·td:nth-child(2)`⏎······`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      const secondChild = document.querySelector(
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,80 @@
+};
+
+let previousBtn: HTMLElement | any = document.querySelector(
+  "button[data-prevbtn]");
```
Codacy has a fix for the issue: [Insert `⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "button[data-prevbtn]"
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,80 @@
+let nextBtn: HTMLElement | any = document.querySelector(
+  "button[data-nextbtn]");
+let pageView: HTMLElement | any = document.querySelector(
+  "label[data-pageview]");
```
Codacy has a fix for the issue: [Insert `⏎`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "label[data-pageview]"
```

0xtruly committed:
fix failing test

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+      tR?.setAttribute("data-entryid", item.id);
+      const firstChild = document.querySelector(
+        `tbody > tr:nth-child(${index}) > td:nth-child(1)`
+        );
```
Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      );
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+"button[data-nextbtn]"
+);
+let pageView: HTMLElement | any = document.querySelector(
+"label[data-pageview]"
```
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "label[data-pageview]"
```

dekunleroy replied:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "label[data-pageview]"
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+};
+
+let previousBtn: HTMLElement | any = document.querySelector(
+"button[data-prevbtn]"
```
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "button[data-prevbtn]"
```

dekunleroy replied:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "button[data-prevbtn]"
```

0xtruly committed:
little fix

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,89 @@
+      tR?.setAttribute("data-entryid", item.id);
+      const firstChild = document.querySelector(
+        `tbody > tr:nth-child(${index}) > td:nth-child(1)`
+        );
```
Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
      );
```

0xtruly committed:
little fix

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,87 @@
+};
+
+let previousBtn: HTMLElement | any = document.querySelector(
+"button[data-prevbtn]"
```
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "button[data-prevbtn]"
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,87 @@
+      previousBtn.style.color = "black";
+      previousBtn.style.backgroundColor = "gray";
+    }
+    console.log('resData', resData)
```
Codacy has a fix for the issue: [Replace `'resData',·resData)` with `"resData",·resData);`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    console.log("resData", resData);
```

dekunleroy commented on src/app.ts:
```diff
@@ -1,6 +1,87 @@
+"button[data-nextbtn]"
+);
+let pageView: HTMLElement | any = document.querySelector(
+"label[data-pageview]"
```
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
  "label[data-pageview]"
```

0xtruly committed:
Use querySelctorAll to access table rows instead of using querySelector
replace innerHTML with textContent for better performance, as innerHTML parses content as HTML which is slower than plain text

0xtruly commented on src/app.ts:
```diff
@@ -1,6 +1,73 @@
+    }
+
+    const tableRow = document.querySelectorAll(`tbody[data-sink] > tr`);
+    console.log('tableRow', tableRow[0].children)
```
Codacy has a fix for the issue: [Replace `'tableRow',·tableRow[0].children)` with `"tableRow",·tableRow[0].children);`](https://app.codacy.com/gh/dekunleroy/pipeline_assessment/pullRequest?prid=9962768)
```suggestion
    console.log("tableRow", tableRow[0].children);
```
